### ISSUE TITLE
upsample_bilinear backward is supper slow in bf16 as compared to fp32

### ISSUE THREAD

**Issue by mingxiao2** [pytorch/pytorch#140096]
### 🐛 Describe the bug

backward of upsample_bilinear is supper slow in bf16 as compared to fp32 

following is the test case:
import torch
import torch.nn as nn
import time

# 简单的自定义模型示例，包含upsample_bilinear2d操作
class MyModel(nn.Module):
    def __init__(self):
        super(MyModel, self).__init__()
        self.conv = nn.Conv2d(3, 1024, 3, padding=1)
        self.upsample = nn.Upsample(scale_factor=2, mode='bilinear', align_corners=True)

    def forward(self, x):
        x = self.conv(x)
        x = self.upsample(x)
        return x

def profile_forward_backward(model, input_data, dtype):
    model = model.to('cuda')
    input_data = input_data.to('cuda').to(dtype)
    optimizer = torch.optim.SGD(model.parameters(), lr=0.01)

    start_time = time.time()
    for _ in range(1000):  # 多次迭代以获得更稳定的时间测量
        optimizer.zero_grad()
        with torch.autocast(device_type='cuda', dtype=dtype):
            output = model(input_data)
            loss = output.mean()
        loss.backward()
        optimizer.step()
    end_time = time.time()
    elapsed_time = end_time - start_time
    print(f"{dtype} - Forward and backward time: {elapsed_time} seconds")

# 创建模型和输入数据
model = MyModel()
input_data = torch.randn(1, 3, 32, 32)

# 使用bf16进行性能分析
profile_forward_backward(model, input_data, torch.bfloat16)

# 使用fp32进行性能分析
profile_forward_backward(model, input_data, torch.float32)

### Versions

pytorch2.3.0-py3.10-cudnn8-cuda12.3

**Comment by mingxiao2** [comment]
> What GPU are you running this on?
 NVIDIA A100-SXM, 80G

**Comment by mingxiao2** [comment]
@malfet may you please help double check this issue? Thanks in advance.

**Comment by mingxiao2** [comment]
intersting, we can see such big performance gap on NGC 23.10-py3, but the gap is quite close on NGC 24.10-py3
1) NGC 24.10-py3
finish warm-up
torch.bfloat16 - Forward and backward time: 0.5524475574493408 seconds
finish warm-up
torch.float32 - Forward and backward time: 0.45501065254211426 seconds

2)NGC 23.10-py3
finish warm-up
torch.bfloat16 - Forward and backward time: 5.039268732070923 seconds
finish warm-up
torch.float32 - Forward and backward time: 0.5534732341766357 seconds
